app/src/lib/stores/helpers/infer-compare-to-branch.ts

```diff
@@ -0,0 +1,153 @@
+import { Branch, IAheadBehind } from '../../../models/branch'
+import { PullRequest } from '../../../models/pull-request'
+import { getAheadBehind, revRange } from '../../git'
+import { Repository } from '../../../models/repository'
+import { GitHubRepository } from '../../../models/github-repository'
+
+interface IOfGithub {
+  kind: 'github'
+  repository: Repository
+}
+
+interface IOfPullRequest {
+  kind: 'pr'
+  pullRequest: PullRequest
+}
+
+interface IOfFork {
+  kind: 'fork'
+  repository: Repository
+  gitHubRepository: GitHubRepository
+  currentBranch: Branch
+}
+
+// I am struggling to come up with a decent name for this type
+type NamingIsHard = IOfPullRequest | IOfGithub | IOfFork
```

Comment:
I don't have a good names for this type nor the 3 interfaces that compose it.

I'm using this type to make it easier for the caller to make use of the function, `inferComparisonBranch`. Leaving it as a list of arguments would require users of this function to have to concern themselves with the implementation details of this function, something I'm looking to avoid. I thought of this fairly quickly, so I'm sure this could be done better. Open to suggestion 👍 

Thoughts @desktop/core

Reply:
When would you need to call this function without knowing what type of object you’re passing? Would it make sense to just export the three more specific functions and call them individually?

Reply:
> When would you need to call this function without knowing what type of object you’re passing? 

The only case you wouldn't need to know what type the object is, is if you have a non-published repository. In all other case, you'd need to know. 

> Would it make sense to just export the three more specific functions and call them individually?

It would. In fact, that's how I originally had it, but doing it this way ensures the caller doesn't have to handle all of that logic themselves. The reason why this is important is because these checks are ordinal. It really becomes a question of _what level do I want to perform these checks?_

As it stands, I think doing it here is the best since it ensures we only have to care about one function that can delegate the function calls without needing to add yet another layer of abstraction.

Do you feel like this function is doing a bit too much or are you thinking having the caller construct one of these types is the wrong approach?